.github/workflows/sycl-linux-precommit.yml

```diff
@@ -219,7 +219,84 @@ jobs:
+
+  E2E-with-new-offload-model:
+    needs: [build, detect_changes, compat_read_exclude]
+    if: |
+      !cancelled() &&
+      needs.build.outputs.build_conclusion == 'success' &&
+      contains(github.event.pull_request.labels.*.name, 'new-offload-model')
+    permissions:
+      contents: write
+      packages: read
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Intel / GEN 12 Integrated
+            runner: '["Linux", "gen12"]'
+            target_devices: level_zero:gpu;opencl:gpu;opencl:cpu
+          - name: NVIDIA/CUDA
+            runner: '["Linux", "cuda"]'
+            image_options: -u 1001 --gpus all --cap-add SYS_ADMIN
+            target_devices: cuda:gpu
+          - name: AMD/HIP
+            runner: '["Linux", "amdgpu"]'
+            image_options: -u 1001 --device=/dev/dri --device=/dev/kfd
+            target_devices: hip:gpu
+            extra_lit_opts: -j 1
+          - name: Intel / Arc A-Series Graphics
+            runner: '["Linux", "arc"]'
+            target_devices: level_zero:gpu;opencl:gpu;level_zero_v2:gpu
+          - name: Intel Dev IGC / Arc A-Series Graphics
+            runner: '["Linux", "arc"]'
+            image: ghcr.io/intel/llvm/ubuntu2404_intel_drivers:devigc
+            target_devices: level_zero:gpu
+            use_igc_dev: true
+            env: '{"LIT_FILTER":"Matrix/"}'
+          - name: Intel / Ponte Vecchio GPU
+            runner: '["Linux", "pvc"]'
+            target_devices: level_zero:gpu;opencl:gpu;level_zero_v2:gpu
+          - name: Intel Dev IGC / Ponte Vecchio GPU
+            runner: '["Linux", "pvc"]'
+            image: ghcr.io/intel/llvm/ubuntu2404_intel_drivers:devigc
+            target_devices: level_zero:gpu
+            use_igc_dev: true
+            env: '{"LIT_FILTER":"Matrix/"}'
+          - name: Intel / Battlemage Graphics
+            runner: '["Linux", "bmg"]'
+            target_devices: level_zero_v1:gpu;level_zero_v2:gpu
+
+    uses: ./.github/workflows/sycl-linux-run-tests.yml
+    with:
+      name: ${{ matrix.name }} with NewOffloadModel
+      runner: ${{ matrix.runner }}
+      image: ${{ matrix.image }}
+      image_options: ${{ matrix.image_options || '-u 1001 --device=/dev/dri -v /dev/dri/by-path:/dev/dri/by-path --privileged --cap-add SYS_ADMIN' }}
+      target_devices: ${{ matrix.target_devices }}
+      extra_lit_opts: ${{ matrix.extra_lit_opts }} " --param enable_new_offload_model=True"
+      repo_ref: ${{ github.sha }}
+      toolchain_artifact: ${{ needs.build.outputs.toolchain_artifact }}
+      toolchain_artifact_filename: ${{ needs.build.outputs.toolchain_artifact_filename }}
+      toolchain_decompress_command: ${{ needs.build.outputs.toolchain_decompress_command }}
+      binaries_artifact: ${{ matrix.binaries_artifact || 'e2e_bin_with_new_offload_model' }}
+      testing_mode: ${{ matrix.testing_mode || 'run-only' }}
```

Comment:
Question: new-offload-driver option should be passed on a build stage, not run stage. Should we first build with new offload driver and then run? I see that run part is here, but where is the build part of the new offload driver testing?